mods/stairs/init.lua

```diff
@@ -233,6 +235,121 @@ function stairs.register_slab(subname, recipeitem, groups, images, description,
 end
 
 
+-- Register corners
+-- Node will be called stairs:stairi_<subname>
+
+function stairs.register_stairi(subname, recipeitem, groups, images, description, sounds)
+	minetest.register_node(":stairs:stairi_" .. subname, {
+		description = description,
+		drawtype = "nodebox",
+		tiles = images,
+		paramtype = "light",
+		paramtype2 = "facedir",
+		is_ground_content = false,
+		groups = groups,
+		sounds = sounds,
+		node_box = {
+			type = "fixed",
+			fixed = {
+				{-0.5, -0.5, -0.5, 0.5, 0, 0.5},
+				{-0.5, 0, 0, 0.5, 0.5, 0.5},
+                {-0.5, 0.5, -0.5, 0, 0, 0},
+			},
+		},
+		on_place = function(itemstack, placer, pointed_thing)
+			if pointed_thing.type ~= "node" then
+				return itemstack
+			end
+
+			return rotate_and_place(itemstack, placer, pointed_thing)
+		end,
+	})
+
+	-- for replace ABM
+	if replace then
+		minetest.register_node(":stairs:stairi_" .. subname .. "upside_down", {
```

Comment:
I did use copy/paste, then went in and hand-typed some changes. sorry, my first PR and I thought I was only doing a PR to my own fork and something went wrong : /